lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -1747,6 +1747,41 @@ ::mlir::LogicalResult CollectivePermuteOp::verify() {
   return success();
 }
 
+::mlir::OpFoldResult
+mlir::tt::ttnn::CollectivePermuteOp::fold(FoldAdaptor adaptor) {
+  ::mlir::DenseIntElementsAttr srcTargetPairs = getSourceTargetPairs();
+
+  // Filter out self-mapping src-target pairs
+  SmallVector<APInt> filteredPairs;
+  for (auto it = srcTargetPairs.begin(), e = srcTargetPairs.end(); it != e;) {
+    auto src = *it++;
+    auto target = *it++;
+    if (src == target) {
+      continue;
+    }
+    filteredPairs.push_back(src);
+    filteredPairs.push_back(target);
+  }
```

Comment:
I think you can rewrite this without iterators
```suggestion
  auto elements = srcTargetPairs.getValues<APInt>();
  SmallVector<APInt> filteredPairs;
  for (size_t i = 0; i < elements.size(); i += 2) {
    auto src = elements[i];
    auto target = elements[i + 1];
    if (src == target) {
      continue;
    }
    filteredPairs.push_back(src);
    filteredPairs.push_back(target);
  }
```

Reply:
Is this suggestion for better readability? Anyway, I updated code as you suggested.

Reply:
> Is this suggestion for better readability?

I would say yes, having indices vs iterators and it's more idiomatic loop.